elfi/methods/diagnostics.py

```diff
@@ -0,0 +1,260 @@
+    and the minimum-MRSSE combination is chosen as the one with the optimal performance.
+
+    Attributes
+    ----------
+    list_ss : array_like
+        List of the to-be-assessed summary statistics.
+    simulator : elfi.Simulator
+        Simulator fit with the observations of the true/underlying model.
+    fn_distance : str or function
+        Function for calculating distance (can be a string if the function is implemented in ELFI).
+    seed : int, optional
+
+    References
+    ----------
+    [1] Nunes, M. A., & Balding, D. J. (2010).
+    On optimal selection of summary statistics for approximate Bayesian computation.
+    Statistical applications in genetics and molecular biology, 9(1).
+    [2] Blum, M. G., Nunes, M. A., Prangle, D., & Sisson, S. A. (2013).
+    A comparative review of dimension reduction methods in approximate Bayesian computation.
+    Statistical Science, 28(2), 189-208.
+
+    """
+
+    def __init__(self, list_ss, simulator, fn_distance, seed=0):
+        """Initialise the summary-statistics selection based on the Two-Stage Procedure.
+
+        Parameters
+        ----------
+        list_ss : array_like
+            List of the to-be-assessed summary statistics.
+        simulator : elfi.Simulator
+            Simulator fit with the observations of the true/underlying model.
+        fn_distance : str or function
+            Function for calculating distance
+            (can be a string if the function is implemented in ELFI).
+        seed : int, optional
+
+        """
+        self.list_ss = list_ss
+        self.simulator = simulator
+        self.fn_distance = fn_distance
+        self.seed = seed
+
+    def run(self, k, n_sim, n_acc, n_closest):
+        """Run the Two-Stage Procedure.
+
+        Parameters
+        ----------
+        k : int
+            Parameter for the kth-nearest-neighbour search.
+        n_sim : int
+            Number of the total ABC-rejection simulations.
+        n_acc : int
+            Number of the accepted ABC-rejection simulations.
+        n_closest : int
+            Number of the `closest' datasets.
+
+        Returns
+        -------
+        array_like
+            Set of the summary statistics showing the optimal performance.
+
+        """
+        # Obtain the combinations of the summary statistics.
+        sets_candidate_ss = self._create_ss_combinations(max_cardinality=k)
+
+        # Find the summary statistics combination with the minimum entropy,
+        # preserve the parameters (thetas) corresponding to the `closest' datasets.
+        thetas = {}
+        E_me = np.inf
+        for set_ss in sets_candidate_ss:
+            names_ss = [ss.__name__ for ss in set_ss]
+            thetas_ss = self._obtain_accepted_thetas(set_ss, n_sim, n_acc)
+            thetas[set_ss] = thetas_ss
+            E_ss = self._calc_entropy(thetas_ss, n_acc, k)
+            if E_ss < E_me:
```

Comment:
Additionally you could keep track of some number of best performing combinations to speed up stage 2.

Reply:
Do we want to have more options which have or could be set by the user? At this point, It might unnecessarily overcomplicate the diagnostics procedure as the diagnostics do not have a computational overhead.